third-party/build-third-party.sh

```diff
@@ -0,0 +1,129 @@
+#! /usr/bin/env bash
+#
+# Copyright (c) 2019 vesoft inc. All rights reserved.
+#
+# This source code is licensed under Apache 2.0 License,
+# attached with Common Clause Condition 1.0, found in the LICENSES directory.
+
+
+# Use CC and CXX environment variables to use custom compilers.
+
+start_time=$(date +%s)
+
+# Always use bash
+shell=$(basename $(readlink /proc/$$/exe))
+if [ ! x$shell = x"bash" ]
+then
+    bash $0 $@
+    exit $?
+fi
+
+# CMake and GCC version checking
+function version_cmp {
+    mapfile -t left < <( echo $1 | tr . '\n' )
+    mapfile -t right < <( echo $2 | tr . '\n')
+    local i
+    for i in ${!left[@]}
+    do
+        local lv=${left[$i]}
+        local rv=${right[$i]}
+        [[ -z $rv ]] && { echo $lv; return; }
+        [[ $lv -ne $rv ]] && { echo $((lv - rv)); return; }
+    done
+    ((i++))
+    rv=${right[$i]}
+    [[ ${#right[@]} -gt ${#left[@]} ]] && { echo $((0-rv)); return; }
+}
+
+function check_cmake {
+    hash cmake &> /dev/null || { echo "No cmake found." 1>&2; exit 1; }
+    local cmake_version=$(cmake --version | head -1 | cut -d ' ' -f 3)
+    local least_cmake_version=3.5.0
+    if [[ $(version_cmp $cmake_version $least_cmake_version) -lt 0 ]]
+    then
+        echo "cmake $least_cmake_version or later required" 1>&2
```

Comment:
Why not run 'build-cmake.sh'

Reply:
Hoping to give users a chance to install or use their own CMake installation.